department-of-veterans-affairs/content-build
# VACMS-11637: Sets allow by default in robots.txt

## Description

Closes https://github.com/department-of-veterans-affairs/va.gov-cms/issues/11637   

## Approval for this change
https://dsva.slack.com/archives/CBU0KDSB1/p1668711036321659?thread_ts=1666819314.703079&cid=CBU0KDSB1

## Testing done & Screenshots

Test performed on prod prior to change using an online [google/robotstxt checker](https://tamethebots.com/tools/robotstxt-checker):
<img width="1543" alt="Screen Shot 2022-12-09 at 12 43 11 PM" src="https://user-images.githubusercontent.com/221539/206793745-54e53a28-d50e-4d08-ad5f-842d75744cdd.png">

Test performed on local after making the change, using a local build of the same google/robotstxt checker:
<img width="941" alt="Screen Shot 2022-12-09 at 12 45 21 PM" src="https://user-images.githubusercontent.com/221539/206793730-2b5bd3d2-b967-4843-a914-39bfe29b3a8e.png">

## QA steps
Testing robots.txt is easy once deployed, but requires a dev to check sites that are not public facing (like our RI's).
1. Using the google/robotstxt checker
   - [ ] Validate that a once `Disallowed` path is now `Allowed` when using the `usasearch` user agent. An example is the jobs page at https://www.va.gov/jobs. This url is not allowed to index via usasearch user agent on prod currently, but after this is released, the page should be then allowed.

Example test:
```sh
▶ BAZEL_CXXOPTS="-std=c++14" bazel run robots_main -- ~/projects/content-build/src/site/assets/robots.txt usasearch https://va.gov/event
WARNING: Download from https://mirror.bazel.build/github.com/bazelbuild/rules_cc/archive/b7fe9697c0c76ab2fd431a891dbb9a6a32ed7c3e.zip failed: class java.io.FileNotFoundException GET returned 404 Not Found
DEBUG: Rule 'com_google_absl' indicated that a canonical reproducible form can be obtained by modifying arguments sha256 = "5f457197b91598fb76d7db20110102e4460d60f3fa7608ed5d37f99d8b687d5f"
DEBUG: Repository com_google_absl instantiated at:
  /Users/danielsasser/projects/test/robotstxt/WORKSPACE:11:13: in <toplevel>
Repository rule http_archive defined at:
  /private/var/tmp/_bazel_danielsasser/a7d52ebe2750ec3a526951db27eff095/external/bazel_tools/tools/build_defs/repo/http.bzl:355:31: in <toplevel>
DEBUG: Rule 'bazel_skylib' indicated that a canonical reproducible form can be obtained by modifying arguments sha256 = "8373bc65ed7c3dba120d70b64b5ce4a66e91d428d07f0422a2f51ecdfc9cfc54"
DEBUG: Repository bazel_skylib instantiated at:
  /Users/danielsasser/projects/test/robotstxt/WORKSPACE:5:13: in <toplevel>
Repository rule http_archive defined at:
  /private/var/tmp/_bazel_danielsasser/a7d52ebe2750ec3a526951db27eff095/external/bazel_tools/tools/build_defs/repo/http.bzl:355:31: in <toplevel>
INFO: Analyzed target //:robots_main (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
Target //:robots_main up-to-date:
  bazel-bin/robots_main
INFO: Elapsed time: 0.073s, Critical Path: 0.00s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
INFO: Running command line: bazel-bin/robots_main /Users/danielsasser/projects/content-build/src/site/assets/robots.txt usasearch htt
INFO: Build completed successfully, 1 total action
user-agent 'usasearch' with URI 'https://va.gov/event': ALLOWED
```


## Acceptance criteria

- [ ] Update codeowners file to reflect that vsa-frontend now own [Robots.txt (content-build)](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/site/assets/robots.txt)
- [ ] Robots.txt is updated to allow by default

## Definition of done
